### PR TITLE
Docker Support + CI

### PR DESCRIPTION
**Pull request by BrianPugh**

### CODE REVIEW

**Comment by BrianPugh** [comment]
@cHunter789 I'm committing changes over here, might take a few tries to get the CI right.

**Comment by BrianPugh** [comment]
@cHunter789 alright, I got the CI working and the prebuilt image is located at https://hub.docker.com/repository/docker/brianpugh/game-and-watch-patch/general

What are your thoughts on the changes?

**Review by cHunter789 — COMMENTED** [review]
Hi, check the comments. After I fixed the upper case issue all looks good and working for me.

File: docker/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+version: "3"
+services:
+  game-and-watch-patch:
+    container_name: game-and-watch-patch
+    image: BrianPugh/game-and-watch-patch:latest
```
[the remark this answers is not in the document]
Comment (BrianPugh) [reply docker/docker-compose.yaml]: addressed in [e7a8f8c](https://github.com/BrianPugh/game-and-watch-patch/pull/58/commits/e7a8f8cbf5a01e91365e52cb42875c39bb2f3bd5)

File: README.md [hunk 2]
```diff
@@ -88,7 +88,37 @@ make clean
...
+  This Dockerfile is written tageting an x86-64 machine running Linux.
+
+  Steps to flash from a docker container (running on Linux, e.g. Archlinux or Ubuntu):
+
+  ```bash
+  # Go into the docker directory of this repo.
+  cd docker/
+
+  # Pull the pre-built docker image.
+  docker pull game-and-watch-patch:latest
```
[the remark this answers is not in the document]
Comment (BrianPugh) [reply README.md]: addressed in [e7a8f8c](https://github.com/BrianPugh/game-and-watch-patch/pull/58/commits/e7a8f8cbf5a01e91365e52cb42875c39bb2f3bd5)

**Comment by BrianPugh** [comment]
thanks for the feedback! merging now.

**Pull request merged**